zebra/zebra_vxlan.c

```diff
@@ -3513,13 +3520,16 @@ void zebra_vxlan_dup_addr_detection(ZAPI_HANDLER_ARGS)
 	STREAM_GETL(s, freeze);
 	STREAM_GETL(s, freeze_time);
 
+	old_addr_detect = zebra_evpn_do_dup_addr_detect(zvrf);
+	zvrf->dup_addr_detect = dup_addr_detect;
+	dup_addr_detect = zebra_evpn_do_dup_addr_detect(zvrf);
+
 	/* DAD previous state was enabled, and new state is disable,
 	 * clear all duplicate detected addresses.
 	 */
-	if (zvrf->dup_addr_detect && !dup_addr_detect)
+	if (old_addr_detect && !dup_addr_detect)
 		zebra_vxlan_clear_dup_detect_vni_all(zvrf);
```

Comment:
When DAD is enabled, does the flag `ZEBRA_EVPN_MH_DUP_ADDR_DETECT_OFF` need to unset?

Reply:
ZEBRA_EVPN_MH_DUP_ADDR_DETECT_OF is unset/off by default and set when the first local ES is configured. There are other MH resources are bootstraped this way i.e. on first ES config as there is no global knob for disabling/enabling MH. Once MH is turned on you will have to remove the config and restart FRR to turn it off.